lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -0,0 +1,30 @@
+# typed: false
+# frozen_string_literal: true
+
+require "json"
+
+running = true
+while running
+  # Read headers until line breaks
+  headers = $stdin.gets("\r\n\r\n")
+
+  # Read the response content based on the length received in the headers
+  request = $stdin.read(headers[/Content-Length: (\d+)/i, 1].to_i)
+
+  json = JSON.parse(request, symbolize_names: true)
+
+  request_method = json.fetch(:method)
+  params = json[:params]
+
+  response_json = nil
+  case request_method
+  when "shutdown"
+    # returning the column names just to illustrate we have access the Rails app
+    response_json = { result: "ok", columns: User.column_names }.to_json
+
```

Comment:
I think this will make the prototype confusing to judge. Let's keep the `shutdown` as the endpoint that finishes the loop and add a separate one surfacing the column information.

It doesn't need to be connected to the addon part, but having it separate paints a clearer picture of the approach.